.github/workflows/sycl-containers.yaml

```diff
@@ -51,6 +52,51 @@ jobs:
             file: ubuntu2404_build
             tag: latest
             build_args: ""
+    steps:
+      # Cleanup runner before building containers, as sometimes it
+      # runs out of space during image build.
+      - name: Cleanup runner
+        run: |
+          echo "Available space before:"
+          df -hT
+
+          rm -rf /opt/hostedtoolcache
+          rm -rf /usr/share/dotnet/
+          rm -rf /usr/share/swift
+          sudo rm -rf /usr/local/share/powershell
+          sudo rm -rf /usr/local/share/chromium
+          sudo rm -rf /usr/local/lib/android
+
+          echo "Available space after:"
+          df -hT
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 2
+      - name: Build and Push Container
+        uses: ./devops/actions/build_container
+        with:
+          push: ${{ github.event_name != 'pull_request' }}
+          file: ${{ matrix.file }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          tags: |
+            ghcr.io/${{ github.repository }}/${{ matrix.file }}:${{ matrix.tag }}-${{ github.sha }}
+            ghcr.io/${{ github.repository }}/${{ matrix.file }}:${{ matrix.tag }}
+          build-args: ${{ matrix.build_args }}
+
+  # Then build "Intel Drivers" images that depend on previous images.
+  # Note: Building these images on PR means using old "Base" and "Build" images,
+  # as the ones above were not yet pushed.
```

Comment:
Isn't it a deal breaker for us? @sarnex , @uditagarwal97

Reply:
+1 to just moving docker container build jobs to self-hosted runners

Reply:
if you want to move building dockers to self-hosted runners, pls let me know which label should I used, at best.

Reply:
I'm trying to use base/build image after creation to be a base for "intel drivers" image. In theory this should work since we use self-hosted runner and `docker images` sees it... but perhaps this can't be simply done...

Should we not care about this issue and go with what I already have (split jobs in 2 parts + change to self-hosted runner)?